rest/rule_test.go

```diff
@@ -987,6 +987,23 @@ func TestApplyInboundRulesNestedRulesSlice(t *testing.T) {
 	assert.Nil(err, "Error should be nil")
 }
 
+// Ensures that nested inbound Rules are correctly applied to slices of primitives.
+func TestApplyNestedInboundRulesSliceOfPrimitive(t *testing.T) {
+	assert := assert.New(t)
+	payload := []string{"foo", "bar"}
+	rules := NewRules((*string)(nil),
+		&Rule{
+			Type: String,
+		},
+	)
+
+	actual, err := applyNestedInboundRules(payload, rules, "1")
+
+	assert.Equal([]interface{}{"foo", "bar"}, actual, "Incorrect return value")
+
+	assert.Nil(err, "Error should be nil")
+}
+
```

Comment:
Should probably have a test case which calls `applyInboundRules` with a nested payload and rules.

Also should test that this appropriately fails if the payload slice contains values which cannot be coerced to the rule type.